components/brave_wallet/common/eth_sign_typed_data_helper.cc

```diff
@@ -0,0 +1,327 @@
+    CHECK(base::HexStringToBytes(value_str->data() + 2, &address));
+    DCHECK_EQ(address.size(), 20u);
+    for (size_t i = 0; i < 256 - 160; i += 8)
+      result.push_back(0);
+    result.insert(result.end(), address.begin(), address.end());
+
+  } else if (base::StartsWith(type, "bytes", base::CompareCase::SENSITIVE)) {
+    unsigned type_check;
+    if (!base::StringToUint(type.data() + 5, &type_check) || type_check > 32)
+      return absl::nullopt;
+    const std::string* value_str = value.GetIfString();
+    if (!value_str || !IsValidHexString(*value_str))
+      return absl::nullopt;
+    std::vector<uint8_t> bytes;
+    CHECK(base::HexStringToBytes(value_str->data() + 2, &bytes));
+    if (bytes.size() > 32)
+      return absl::nullopt;
+    result.insert(result.end(), bytes.begin(), bytes.end());
+    for (size_t i = 0; i < 32u - bytes.size(); ++i) {
+      result.push_back(0);
+    }
+  } else if (base::StartsWith(type, "uint", base::CompareCase::SENSITIVE)) {
+    // uint8 to uint256 in steps of 8
+    unsigned type_check;
+    if (!base::StringToUint(type.data() + 4, &type_check) || type_check % 8 ||
+        type_check > 256)
+      return absl::nullopt;
+
+    absl::optional<int> value_int = value.GetIfInt();
+    if (!value_int)
+      return absl::nullopt;
+    uint256_t encoded_value = *value_int;
+    // check if value excceeds type bound
+    switch (type_check) {
+      case 8:
+        if (encoded_value > std::numeric_limits<uint8_t>::max())
+          return absl::nullopt;
+        break;
+      case 16:
+        if (encoded_value > std::numeric_limits<uint16_t>::max())
+          return absl::nullopt;
+        break;
+      case 32:
+        if (encoded_value > std::numeric_limits<uint32_t>::max())
+          return absl::nullopt;
+        break;
+      default:
+        // TODO(darkdh): base::Value needs to support uint256 type
+        break;
+    }
+    for (int i = 256 - 8; i >= 0; i -= 8) {
+      result.push_back((encoded_value >> i) & 0xFF);
+    }
+  } else if (base::StartsWith(type, "int", base::CompareCase::SENSITIVE)) {
+    // int8 to int256 in steps of 8
+    unsigned type_check;
+    if (!base::StringToUint(type.data() + 3, &type_check) || type_check % 8 ||
+        type_check > 256)
+      return absl::nullopt;
+    absl::optional<int> value_int = value.GetIfInt();
+    if (!value_int)
+      return absl::nullopt;
+    int256_t encoded_value = *value_int;
+    // check if value excceeds type bound
+    switch (type_check) {
+      case 8:
+        if (encoded_value > std::numeric_limits<int8_t>::max())
+          return absl::nullopt;
+        break;
+      case 16:
+        if (encoded_value > std::numeric_limits<int16_t>::max())
+          return absl::nullopt;
+        break;
+      case 32:
+        if (encoded_value > std::numeric_limits<int32_t>::max())
+          return absl::nullopt;
+        break;
+      default:
+        // TODO(darkdh): base::Value needs to support int256 type
```

Comment:
nit: pls make comment say int64_t int128_t and int256_T

Reply:
fixed in 6bbb1c4516cb09859c699807dc35b5fd17d06b54